src/main.rs

```diff
@@ -1,3 +1,19 @@
+#![deny(clippy::all)]
+#![deny(clippy::pedantic)]
+#![allow(
+    clippy::new_without_default,
+    // TODO: Remove when fixed
+	clippy::result_unit_err,
+    clippy::default_trait_access,
+    clippy::missing_errors_doc,
+    clippy::missing_panics_doc,
+    clippy::implicit_hasher,
+    clippy::too_many_lines,
+    // More explicit sometimes to have the module name
+    clippy::module_name_repetitions
+)]
+#![warn(clippy::cast_precision_loss, clippy::cast_possible_truncation, clippy::cast_sign_loss)]
```

Comment:
I searched a bit, and you're right, it can be defined at workspace level, so I moved everything!

Even if 95% was working, those 3 are not applied for a reason I don't get, so for now I put them back at lib level:
- clippy::new_without_default
- clippy::too_many_lines
- clippy::result_unit_err